server/tests/controllers/users.js

```diff
@@ -49,9 +50,11 @@ const correctDetails = { email: 'emekag@gmail.com', password: 'emeka' };
 const incorrectDetails = { email: 'emekag@gmail.com', password: 'wrongpassword' };
 const emptyEmailField = { email: '', password: 'emeka' };
 const emptyPasswordField = { email: 'emekag@gmail.com', password: '' };
+const token = jwt.sign({ user: { email: user.email }, links: { reset: 'http://localhost:5000' } }, process.env.JWT_SECRET_TOKEN, { expiresIn: '2h' });
```

Comment:
I'd like to know why links: { reset: 'http://localhost:5000' }

Reply:
We are working on the API for the app and is most likely that the frontend URL may differ from the Backend(API) URL.

In this regard, it's expected that any link that users will have to click should redirect to the frontend part of the app that will now make to relevant api endpoint(s) to process the user's purpose of clicking the link.

This should also apply to the reset link. Since the frontend is not up yet and there is currently no URL for it, also in the case of building different app(each with different URL) with the API, I decided to make provisions for whoever is consuming the recover password endpoint to specify his/her recover link that will be sent to users when they request for new password.

If the frontend link used to make a request to the password recover endpoint is 'https://authorshaven.com/password_reset', recovery link sent to the user's email will be 'https://authorshaven.com/password_reset/{JsonWebToken}'

Reply:
links is an object containing links related to the resource.

Reply:
@d-beloved @